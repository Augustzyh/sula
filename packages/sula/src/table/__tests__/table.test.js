import React from 'react';
import { Table } from '..';
import TableAction from '../TableAction';
import { mount } from 'enzyme';
import { dataSource, actWait } from '../../__tests__/common';

describe('table', () => {
  describe('table snapshot', () => {
    it('basic table', () => {
      const wrapper = mount(
        <Table
          rowKey="id"
          initialDataSource={dataSource}
          columns={[
            {
              key: 'id',
              title: 'Id',
            },
            {
              key: 'name',
              title: 'Name',
              dataIndex: 'name',
            },
            {
              key: 'age',
              title: 'Age',
              render: () => <div>age</div>,
            },
          ]}
          rowSelection={{
            hideDefaultSelections: true,
          }}
          actionsRender={[
            {
              type: 'button',
              props: {
                children: 'right',
              },
            },
          ]}
          leftActionsRender={[
            {
              type: 'button',
              props: {
                children: 'left',
              },
            },
          ]}
        />,
      );
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('no pagination', async () => {
      let wrapper;
      await actWait(() => {
        wrapper = mount(
          <Table
            rowKey="id"
            remoteDataSource={{
              url: '/nopagination.json',
              method: 'post',
            }}
            initialPaging={{
              pagination: false,
            }}
            columns={[
              {
                key: 'id',
                title: 'Id',
              },
              {
                key: 'name',
                title: 'Name',
              },
              {
                key: 'age',
                title: 'Age',
              },
            ]}
            rowSelection={{}}
          />,
        );
      })
      await actWait();
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('no paging', async () => {
      const wrapper = mount(
        <Table
          rowKey="id"
          remoteDataSource={{
            url: '/error.json',
            method: 'post',
          }}
          initialPaging={false}
          columns={[
            {
              key: 'id',
              title: 'Id',
            },
            {
              key: 'name',
              title: 'Name',
            },
            {
              key: 'age',
              title: 'Age',
            },
          ]}
        />,
      );
      await actWait();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('tableAction', () => {
    it('do not have actionsRender will return null', () => {
      const wrapper = mount(<TableAction />);
      expect(wrapper.find('.sula-table-action-left').children()).toEqual({});
      expect(wrapper.find('.sula-table-action-right').children()).toEqual({});
    });

    it('have ctx', () => {
      const fn = jest.fn();
      const wrapper = mount(
        <Table
          rowKey="id"
          initialDataSource={dataSource}
          columns={[
            {
              key: 'id',
              title: 'Id',
            },
            {
              key: 'name',
              title: 'Name',
              dataIndex: 'name',
            },
            {
              key: 'age',
              title: 'Age',
            },
          ]}
          actionsRender={[
            {
              type: 'button',
              props: {
                className: 'jest-test-btn',
                children: 'right',
              },
              action: fn,
            },
          ]}
          leftActionsRender={[
            {
              type: 'button',
              props: {
                children: 'left',
              },
            },
          ]}
        />,
      );

      wrapper.find('.jest-test-btn').last().simulate('click');
      const ctx = fn.mock.calls[0][0];
      expect(ctx.table).not.toBeNull();
      expect(ctx.dataSource).not.toBeNull();
    });
  });
});
