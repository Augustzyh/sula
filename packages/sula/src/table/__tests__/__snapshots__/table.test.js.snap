// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table table snapshot basic table 1`] = `"<div class=\\"ant-table-wrapper\\"><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><div class=\\"ant-table\\"><div class=\\"ant-table-title\\"><div class=\\"sula-table-action-wrapper\\"><div class=\\"sula-table-action-left\\"><button type=\\"button\\" class=\\"ant-btn\\"><span>left</span></button></div><div class=\\"sula-table-action-right\\"><button type=\\"button\\" class=\\"ant-btn\\"><span>right</span></button></div></div></div><div class=\\"ant-table-container\\"><div class=\\"ant-table-content\\"><table style=\\"table-layout: auto;\\"><colgroup><col class=\\"ant-table-selection-col\\"></colgroup><thead class=\\"ant-table-thead\\"><tr><th class=\\"ant-table-cell ant-table-selection-column\\"><div class=\\"ant-table-selection\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></div></th><th class=\\"ant-table-cell\\">Id</th><th class=\\"ant-table-cell\\">Name</th><th class=\\"ant-table-cell\\">Age</th></tr></thead><tbody class=\\"ant-table-tbody\\"><tr data-row-key=\\"0\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">0</td><td class=\\"ant-table-cell\\">lily0</td><td class=\\"ant-table-cell\\"><div>age</div></td></tr><tr data-row-key=\\"1\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">1</td><td class=\\"ant-table-cell\\">lily1</td><td class=\\"ant-table-cell\\"><div>age</div></td></tr><tr data-row-key=\\"2\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">2</td><td class=\\"ant-table-cell\\">lily2</td><td class=\\"ant-table-cell\\"><div>age</div></td></tr><tr data-row-key=\\"3\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">3</td><td class=\\"ant-table-cell\\">lily3</td><td class=\\"ant-table-cell\\"><div>age</div></td></tr><tr data-row-key=\\"4\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">4</td><td class=\\"ant-table-cell\\">lily4</td><td class=\\"ant-table-cell\\"><div>age</div></td></tr><tr data-row-key=\\"5\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">5</td><td class=\\"ant-table-cell\\">lily5</td><td class=\\"ant-table-cell\\"><div>age</div></td></tr><tr data-row-key=\\"6\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">6</td><td class=\\"ant-table-cell\\">lily6</td><td class=\\"ant-table-cell\\"><div>age</div></td></tr><tr data-row-key=\\"7\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">7</td><td class=\\"ant-table-cell\\">lily7</td><td class=\\"ant-table-cell\\"><div>age</div></td></tr><tr data-row-key=\\"8\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">8</td><td class=\\"ant-table-cell\\">lily8</td><td class=\\"ant-table-cell\\"><div>age</div></td></tr><tr data-row-key=\\"9\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">9</td><td class=\\"ant-table-cell\\">lily9</td><td class=\\"ant-table-cell\\"><div>age</div></td></tr></tbody></table></div></div></div><ul class=\\"ant-pagination ant-table-pagination ant-table-pagination-right\\" unselectable=\\"unselectable\\"><li title=\\"Previous Page\\" class=\\"ant-pagination-prev ant-pagination-disabled\\" aria-disabled=\\"true\\"><a class=\\"ant-pagination-item-link\\" disabled=\\"\\"><span role=\\"img\\" aria-label=\\"left\\" class=\\"anticon anticon-left\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"left\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z\\"></path></svg></span></a></li><li title=\\"1\\" class=\\"ant-pagination-item ant-pagination-item-1 ant-pagination-item-active\\" tabindex=\\"0\\"><a>1</a></li><li title=\\"2\\" class=\\"ant-pagination-item ant-pagination-item-2\\" tabindex=\\"0\\"><a>2</a></li><li title=\\"Next Page\\" tabindex=\\"0\\" class=\\"ant-pagination-next\\" aria-disabled=\\"false\\"><a class=\\"ant-pagination-item-link\\"><span role=\\"img\\" aria-label=\\"right\\" class=\\"anticon anticon-right\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"right\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z\\"></path></svg></span></a></li></ul></div></div></div><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><form class=\\"ant-form ant-form-horizontal\\"></form></div></div><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><form class=\\"ant-form ant-form-horizontal\\"></form></div></div>"`;

exports[`table table snapshot no pagination 1`] = `"<div class=\\"ant-table-wrapper\\"><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><div class=\\"ant-table\\"><div class=\\"ant-table-container\\"><div class=\\"ant-table-content\\"><table style=\\"table-layout: auto;\\"><colgroup><col class=\\"ant-table-selection-col\\"></colgroup><thead class=\\"ant-table-thead\\"><tr><th class=\\"ant-table-cell ant-table-selection-column\\"><div class=\\"ant-table-selection\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></div></th><th class=\\"ant-table-cell\\">Id</th><th class=\\"ant-table-cell\\">Name</th><th class=\\"ant-table-cell\\">Age</th></tr></thead><tbody class=\\"ant-table-tbody\\"><tr data-row-key=\\"0\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">0</td><td class=\\"ant-table-cell\\">lily0</td><td class=\\"ant-table-cell\\">10</td></tr><tr data-row-key=\\"1\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">1</td><td class=\\"ant-table-cell\\">lily1</td><td class=\\"ant-table-cell\\">11</td></tr><tr data-row-key=\\"2\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">2</td><td class=\\"ant-table-cell\\">lily2</td><td class=\\"ant-table-cell\\">12</td></tr><tr data-row-key=\\"3\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">3</td><td class=\\"ant-table-cell\\">lily3</td><td class=\\"ant-table-cell\\">13</td></tr><tr data-row-key=\\"4\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">4</td><td class=\\"ant-table-cell\\">lily4</td><td class=\\"ant-table-cell\\">14</td></tr><tr data-row-key=\\"5\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">5</td><td class=\\"ant-table-cell\\">lily5</td><td class=\\"ant-table-cell\\">15</td></tr><tr data-row-key=\\"6\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">6</td><td class=\\"ant-table-cell\\">lily6</td><td class=\\"ant-table-cell\\">16</td></tr><tr data-row-key=\\"7\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">7</td><td class=\\"ant-table-cell\\">lily7</td><td class=\\"ant-table-cell\\">17</td></tr><tr data-row-key=\\"8\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">8</td><td class=\\"ant-table-cell\\">lily8</td><td class=\\"ant-table-cell\\">18</td></tr><tr data-row-key=\\"9\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">9</td><td class=\\"ant-table-cell\\">lily9</td><td class=\\"ant-table-cell\\">19</td></tr><tr data-row-key=\\"10\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">10</td><td class=\\"ant-table-cell\\">lily10</td><td class=\\"ant-table-cell\\">20</td></tr><tr data-row-key=\\"11\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">11</td><td class=\\"ant-table-cell\\">lily11</td><td class=\\"ant-table-cell\\">21</td></tr><tr data-row-key=\\"12\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">12</td><td class=\\"ant-table-cell\\">lily12</td><td class=\\"ant-table-cell\\">22</td></tr><tr data-row-key=\\"13\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">13</td><td class=\\"ant-table-cell\\">lily13</td><td class=\\"ant-table-cell\\">23</td></tr><tr data-row-key=\\"14\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">14</td><td class=\\"ant-table-cell\\">lily14</td><td class=\\"ant-table-cell\\">24</td></tr><tr data-row-key=\\"15\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">15</td><td class=\\"ant-table-cell\\">lily15</td><td class=\\"ant-table-cell\\">25</td></tr><tr data-row-key=\\"16\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">16</td><td class=\\"ant-table-cell\\">lily16</td><td class=\\"ant-table-cell\\">26</td></tr><tr data-row-key=\\"17\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">17</td><td class=\\"ant-table-cell\\">lily17</td><td class=\\"ant-table-cell\\">27</td></tr><tr data-row-key=\\"18\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">18</td><td class=\\"ant-table-cell\\">lily18</td><td class=\\"ant-table-cell\\">28</td></tr><tr data-row-key=\\"19\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-selection-column\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span></label></td><td class=\\"ant-table-cell\\">19</td><td class=\\"ant-table-cell\\">lily19</td><td class=\\"ant-table-cell\\">29</td></tr></tbody></table></div></div></div></div></div></div><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><form class=\\"ant-form ant-form-horizontal\\"></form></div></div><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><form class=\\"ant-form ant-form-horizontal\\"></form></div></div>"`;

exports[`table table snapshot no paging 1`] = `"<div class=\\"ant-table-wrapper\\"><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><div class=\\"ant-table\\"><div class=\\"ant-table-container\\"><div class=\\"ant-table-content\\"><table style=\\"table-layout: auto;\\"><colgroup></colgroup><thead class=\\"ant-table-thead\\"><tr><th class=\\"ant-table-cell\\">Id</th><th class=\\"ant-table-cell\\">Name</th><th class=\\"ant-table-cell\\">Age</th></tr></thead><tbody class=\\"ant-table-tbody\\"><tr class=\\"ant-table-placeholder\\"><td colspan=\\"3\\" class=\\"ant-table-cell\\"><div class=\\"ant-empty ant-empty-normal\\"><div class=\\"ant-empty-image\\"><svg class=\\"ant-empty-img-simple\\" width=\\"64\\" height=\\"41\\" viewBox=\\"0 0 64 41\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g transform=\\"translate(0 1)\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><ellipse class=\\"ant-empty-img-simple-ellipse\\" cx=\\"32\\" cy=\\"33\\" rx=\\"32\\" ry=\\"7\\"></ellipse><g class=\\"ant-empty-img-simple-g\\" fill-rule=\\"nonzero\\"><path d=\\"M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z\\"></path><path d=\\"M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z\\" class=\\"ant-empty-img-simple-path\\"></path></g></g></svg></div><p class=\\"ant-empty-description\\">No Data</p></div></td></tr></tbody></table></div></div></div></div></div></div><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><form class=\\"ant-form ant-form-horizontal\\"></form></div></div><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><form class=\\"ant-form ant-form-horizontal\\"></form></div></div>"`;
